### PR TITLE
Remove support for legacy WebSockets

### PR DESCRIPTION
**Pull request by michalwski**
The only supported XMPP over WebSockets is the one defined in  [RFC 7395](https://tools.ietf.org/html/rfc7395)

Also following changes has been made:
* removed unnecessary code from `mod_websockets`
* `component_SUITE` now truly uses WebSocket connection in the `xep_0114_ws` group

### CODE REVIEW

File: test/ejabberd_tests/rebar.config [hunk 1]
```diff
@@ -4,7 +4,7 @@
 {require_otp_vsn, "1[789]"}.
 
 {deps, [
-        {escalus, ".*", {git, "git://github.com/esl/escalus.git", "3169334"}},
+        {escalus, ".*", {git, "git://github.com/esl/escalus.git", {branch, "fix-transport-stop"}}},
```
Comment (studzien) [inline test/ejabberd_tests/rebar.config]: Should this escalus branch be merged first and the commit hash be used?

File: test/ejabberd_tests/tests/component_SUITE.erl [hunk 2]
```diff
@@ -178,9 +178,10 @@ register_two_components(Config) ->
 
 try_registering_with_wrong_password(Config) ->
     %% Given a component with a wrong password
-    CompOpts1 = spec(component1, Config),
+    CompOpts1 = ?config(component1, Config),
     CompOpts2 = lists:keyreplace(password, 1, CompOpts1,
                                  {password, <<"wrong_one">>}),
+    ct:print("~p", [CompOpts2]),
```
Comment (studzien) [inline test/ejabberd_tests/tests/component_SUITE.erl]: Is this necessary?

File: test/ejabberd_tests/tests/component_SUITE.erl [hunk 3]
```diff
@@ -247,7 +248,8 @@ disco_components(Config) ->
 
 register_subdomain(Config) ->
     %% Given one connected component
-    CompOpts1 = spec(component1, Config),
+    CompOpts1 = ?config(component1, Config),
+    ct:print("~p", [CompOpts1]),
```
Comment (studzien) [inline test/ejabberd_tests/tests/component_SUITE.erl]: Is this necessary?

File: test/ejabberd_tests/rebar.config (hunk 1 above)
Comment (michalwski) [inline test/ejabberd_tests/rebar.config]: Yes it should :)

File: test/ejabberd_tests/tests/component_SUITE.erl (hunk 2 above)
Comment (michalwski) [inline test/ejabberd_tests/tests/component_SUITE.erl]: A leftover, thanks for pointing this out.

File: test/ejabberd_tests/rebar.config (hunk 1 above)
Comment (michalwski) [inline test/ejabberd_tests/rebar.config]: The relevant PR is esl/escalus#123

Comment (studzien) [inline test/ejabberd_tests/rebar.config]: Merged, the commit is ``1d7d3b716``

**Comment by erszcz** [comment]
If `mod_websockets` is not a `gen_mod` anymore why does it still start with `mod_`? This breaks the convention and therefore the internal API consistency. [Names matter](https://www.infoq.com/articles/API-Design-Joshua-Bloch).

(Just to make sure, I'm __not__ nitpicking for the sake of it - it just bit me, even though it's a change from 2016.)

**Pull request merged**